### PR TITLE
[test] Update paraview_check for new release (5.9 and 5.10) and for new Mesa libraries

### PR DESCRIPTION
**Pull request by jfavre**
I removed the check on Vendor because it is redundant anyway with the check for llvmpipe

### CODE REVIEW

**Comment by jfavre** [comment]
addresses https://jira.cscs.ch/browse/AVAILAPPS-12

**Review by vkarak — APPROVED** [review]

**Comment by vkarak** [comment]
@jfavre The test fails on Eiger and Pilatus. Is this expected?

**Comment by jfavre** [comment]
I cannot reproduce the failure of the test on eiger and pilatus, They work fine for me:
reframe -c paraview_check.py -n ParaViewCheck --system pilatus:mc -r
[  PASSED  ] Ran 1/1 test case(s) from 1 check(s) (0 failure(s), 0 skipped)

reframe -c paraview_check.py -n ParaViewCheck --system eiger:mc -r
[  PASSED  ] Ran 1/1 test case(s) from 1 check(s) (0 failure(s), 0 skipped)

**Comment by vkarak** [comment]
@jfavre It doesn't pass for me in a fresh login. I guess that you have a `ParaView` module already available in your `MODULEPATH`. The test passes if I load `cpeCray` first by doing `-S modules=cpeCray,ParaView`. I think that the `valid_prog_environs` should be `['cpeCray']` on Eiger/Pilatus and not `builtin`. Check the other applications.

File: cscs-checks/apps/paraview/paraview_check.py [hunk 1]
```diff
@@ -9,32 +9,28 @@
...
+    num_tasks = 12
+    num_tasks_per_node = 12
+    modules = ['ParaView']
+    executable = 'pvbatch'
+    executable_opts = ['coloredSphere.py']
+    maintainers = ['JF', 'TM']
+    tags = {'scs', 'production'}
 
-        if self.current_system.name in ['eiger', 'pilatus']:
+    @run_after('setup')
```
Comment (vkarak) [inline cscs-checks/apps/paraview/paraview_check.py:23]: Here is the bug. This should be a post-init hook:

```suggestion
    @run_after('init')
```

The reason for that is that after setup is too late for test filtering; the test has already started running (you can use `skip()` there, but it's much simpler if you just change the hook.

**Review by vkarak — CHANGES_REQUESTED** [review]

**Review by vkarak — APPROVED** [review]

**Review by jfavre — COMMENTED** [review]
looks good. thanks for clarifying

**Pull request merged**